dev-client/src/hooks/appPermissionsHooks.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useContext} from 'react';
+
+import {PermissionResponse} from 'expo-location';
+
+import {ForegroundPermissionsContext} from 'terraso-mobile-client/context/AppPermissionsContext';
+
+// As of Sept 2024, the app permissions hooks supplied by the expo libraries don't trigger component updates
+// when app permissions are updated. Instead, the permissions object stays stale until the next call to
+// get() or request().
+// This hook wraps the library-supplied hooks and should cause components to update when permissions are
+// updated in the app or after they are updated outside the app.
+export type UpdatedPermissionsHookReturnType = [
+  PermissionResponse | null, // permission
+  () => Promise<PermissionResponse>, // request
+  () => Promise<PermissionResponse>, // get
```

Comment:
question: do we ever need to call `get` outside of `AppPermissionsContext`?

Reply:
Is this about what level of the component hierarchy the ForegroundPermissionsProvider should live at, in App.tsx? If so, I'm not 100% sure, but I can't think of anything in particular. And I don't believe anything currently relies on it; the only thing I could think of was GeospatialContext.tsx, which uses state.map.userLocation, which I believe starts null, and only gets updated by BottomTabsScreen if location permissions have been granted.

Lmk if I'm not actually answering your question though!

Reply:
it was more like, "if no one is using this can we keep the API surface smaller". but i'm not sure if that might break usage with `PermissionsRequestWrapper`? not a big deal!